src/main/scala/cromwell/binding/values/WdlFile.scala

```diff
@@ -9,23 +9,38 @@ object WdlFile {
     if (path1.endsWith("/") || path2.startsWith("/")) path1 + path2
     else path1 + "/" + path2
   }
+
+  def apply(value: String, isGlob: Boolean = false): WdlFile = if (isGlob) WdlGlobFile(value) else WdlSingleFile(value)
 }
 
-case class WdlFile(value: String) extends WdlPrimitive {
+sealed trait WdlFile extends WdlPrimitive {
+  val value: String
 
   val wdlType: WdlType = WdlFileType
 
+  def isGlob: Boolean = this match {
+    case x: WdlGlobFile => true
+    case _ => false
+  }
+
   override def add(rhs: WdlValue): Try[WdlValue] = rhs match {
     case r: WdlString => Success(WdlFile(value + r.value))
     case _ => invalid(s"$value + $rhs")
   }
 
   override def equals(rhs: WdlValue): Try[WdlBoolean] = rhs match {
-    case r: WdlFile => Success(WdlBoolean(value.equals(r.value)))
-    case r: WdlString => Success(WdlBoolean(value.toString.equals(r.value)))
+    case r: WdlFile => Success(WdlBoolean(value.equals(r.value) && isGlob.equals(r.isGlob)))
+    case r: WdlString => Success(WdlBoolean(value.toString.equals(r.value.toString) && !isGlob))
     case _ => invalid(s"$value == $rhs")
   }
+  override def valueString = value.toString
+}
 
+case class WdlSingleFile(value: String) extends WdlFile {
   override def toWdlString = "\"" + value.toString + "\""
-  override def valueString = value.toString
 }
+
+case class WdlGlobFile(value: String) extends WdlFile {
```

Comment:
I'm still kind of squeamish about something that will almost certainly expand to reference multiple files being part of a product type that implies a single file, but this may not be the time to address that...